db_test.go

```diff
@@ -256,40 +244,32 @@ func TestOpen_ReadPageSize_FromMeta1_Given(t *testing.T) {
 	// test page size from 1KB (1024<<0) to 16MB(1024<<14)
 	for i := 0; i <= 14; i++ {
 		givenPageSize := 1024 << uint(i)
+		t.Logf("Testing page size %d", givenPageSize)
 		// Create empty database.
-		db := MustOpenWithOption(&bolt.Options{PageSize: givenPageSize})
+		db := btesting.MustCreateDBWithOption(t, &bolt.Options{PageSize: givenPageSize})
 		path := db.Path()
-		defer db.MustClose()
-
-		// Close database.
-		if err := db.DB.Close(); err != nil {
-			t.Fatal(err)
-		}
+		// Close the database
+		db.MustClose()
 
 		// Read data file.
 		buf, err := os.ReadFile(path)
-		if err != nil {
-			t.Fatal(err)
-		}
+		require.NoError(t, err)
 
 		// Rewrite meta pages.
-		meta0 := (*meta)(unsafe.Pointer(&buf[pageHeaderSize]))
-		meta0.pgid++
-		if err := os.WriteFile(path, buf, 0666); err != nil {
-			t.Fatal(err)
+		if i%3 == 0 {
+			t.Logf("#%d: Intentionally corrupt the first meta page for pageSize %d", i, givenPageSize)
+			meta0 := (*meta)(unsafe.Pointer(&buf[pageHeaderSize]))
+			meta0.pgid++
+			err = os.WriteFile(path, buf, 0666)
+			require.NoError(t, err)
 		}
 
 		// Reopen data file.
-		if db, err := bolt.Open(path, 0666, &bolt.Options{PageSize: givenPageSize}); err != nil {
-			t.Fatalf("unexpected error: %s", err)
-		} else {
-			if db.Info().PageSize != givenPageSize {
-				t.Fatalf("The page size is expected to be %d, but actually is %d", givenPageSize, db.Info().PageSize)
-			}
-			if err := db.Close(); err != nil {
-				panic(err)
-			}
-		}
+		db = btesting.MustOpenDBWithOption(t, path, nil)
+		require.Equalf(t, givenPageSize, db.Info().PageSize, "check page size failed")
+		// The db.DB is set to nil in MustClose(), so the testing Cleanup
+		// will do nothing when executing PostTestCleanup().
+		db.PostTestCleanup()
```

Comment:
I would assume db.MustClose() should be executed here, that would the checks and close the db.

In fact - I had better to not export the postTestCleanup method.

Reply:
I thought about execute db.MustClose(). The reason why I did not do it is that it doesn't check the db (db.MustCheck).

Reply:
Anyway, just updated to `db.MustClose()` at the cost of not executing the `db.MustCheck`